src/Storage/Snowflake/SelectSource.php

```diff
@@ -11,38 +11,54 @@ class SelectSource implements SourceInterface, SqlSourceInterface
 {
     private string $query;
 
-    /** @var array<mixed> */
+    /** @var string[] */
     private array $queryBindings;
 
+    /** @var string[] */
+    private array $dataTypes;
+
     /** @var string[] */
     private array $columnsNames;
 
     /** @var string[]|null */
     private ?array $primaryKeysNames = null;
 
     /**
+     * @param string[] $queryBindings
+     * @param string[] $dataTypes
      * @param string[] $columnsNames
      * @param string[]|null $primaryKeysNames
-     * @param array<mixed> $queryBindings
      */
     public function __construct(
         string $query,
         array $queryBindings = [],
+        array $dataTypes = [],
```

Comment:
@zajca toto je ale trochu BC ale musi to byt. Dej vedet az se k tomu dostanes

Reply:
protoze takto mame napsane SelectSource i pro EXA/TD. 

Ve vysledku to chci mit sjednocene , ale zalezi jak to budeme rollovat. 

A - vyhneme se ted BC a udelame to kompatibilni a pripadne BC si nechame na major, kterej prijde trochu pozdeji (kde bychom si pak mohli dovolit BC a toto treba zmenili). Rano na standupu jsi to tak navrhoval. 
B - udelame major rovnou po tomto PR a budeme do connection rollovat rovnou

Imho nevidim benefity A protoze stejne tan major prijde (chceme se zbavit starych interfacu). Jak to vidis?

Reply:
Mno tak bude potřeba ještě `Keboola\Package\StorageBackend\ImportExport\Workspace\SynapseFullLoadImporter` přesunůt sem do IE.

A ještě sem si všiml že máme v connection 
Storage_Service_Backend_SnowflakeImport a Storage_Service_Backend_SnowflakeCopyImport

Reply:
Starého interface sa chceme zbavit, ale vyhnul bych sa big bang v connection, može sa velice lehce něco posrat, něco o tom vím, takže bych volil A roloval to do connection postupně, třeba první pro ABS kde sů teď staré interfaces a pak za pár dní pro S3 kde je stará lib.

Reply:
ok, tak udelam ten parametr posledni aby to nebylo BC a dam si tam TODO aby se to pak sjednotilo pri major updatu